File: src/main/java/io/github/takenoha/towerdefense/paper/RewardQueueReceipt.java
package io.github.takenoha.towerdefense.paper;

import java.util.Objects;
import java.util.UUID;

/** Durable identity copied onto an inventory stack while its queue delivery is being committed. */
public record RewardQueueReceipt(UUID queueId, UUID operationId) {
    public RewardQueueReceipt {
        Objects.requireNonNull(queueId, "queueId");
        Objects.requireNonNull(operationId, "operationId");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/RewardQueueReceipt.kt
package io.github.takenoha.towerdefense.paper

import java.util.Objects
import java.util.UUID
import kotlin.jvm.JvmRecord

/** Durable identity copied onto an inventory stack while its queue delivery is being committed. */
@JvmRecord
data class RewardQueueReceipt(
    val queueId: UUID,
    val operationId: UUID,
) {
    init {
        Objects.requireNonNull(queueId, "queueId")
        Objects.requireNonNull(operationId, "operationId")
    }
}
